src/util/hooks/useTouchModalOutside.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect, useRef } from 'react';
+
+interface UseOutsideClickOptions {
+  onOutsideClick: () => void;
+}
+
+const useTouchModalOutside = ({ onOutsideClick }: UseOutsideClickOptions) => {
+  const containerRef = useRef<HTMLDialogElement>(null);
+  const backgroundRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleOutsideClick = (e: MouseEvent) => {
+      const clickedElement = e.target as Node;
+      const container = containerRef.current;
+      const background = backgroundRef.current;
```

Comment:
안전하게 타입 가드를 사용해서 Node인지 파악하는 것도 좋아보입니다. 타입 강제변환보다는 구체적으로 타입 가드를 사용하는게 좋을 거 같습니다!
```suggestion
    // 안전하게 타입 확인
    if (!(e.target instanceof Node)) return;
    const clickedElement = e.target;
    const container = containerRef.current;
    const background = backgroundRef.current;
```